ssh/ssh.go

```diff
@@ -0,0 +1,85 @@
+package ssh
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// SSHKey represents an SSH private/public key pair for authentication.
+type SSHKey struct {
+	signer    ssh.Signer
+	publicKey ssh.PublicKey
+}
+
+// LoadPrivateKeyFromFile loads an SSH private key from a file.
+// Supports passphrase-protected keys for RSA, ECDSA, and Ed25519 key types.
+func LoadPrivateKeyFromFile(filename, passphrase string) (*SSHKey, error) {
+	if filename == "" {
+		return nil, fmt.Errorf("filename cannot be empty")
+	}
+
+	if strings.Contains(filename, "..") {
```

Comment:
Not sure that it is enough to prevent file path traversal.
Can we remove it and leave `LoadPrivateKeyFromFile` as helper?